cachito/workers/cleanup_job.py

```diff
@@ -23,38 +23,51 @@
 def main():
     """Mark all end of life requests as stale using the REST API."""
     for state in ("complete", "in_progress", "failed"):
-        url = f"{config.cachito_api_url.rstrip('/')}/requests"
-        while url:
-            try:
-                response = session.get(
-                    url, params={"state": state}, timeout=config.cachito_api_timeout
-                )
-            except requests.RequestException:
-                msg = f"The connection failed when querying {url}"
-                log.exception(msg)
-                raise CachitoError(msg)
-
-            if not response.ok:
-                log.error(
-                    "The request to %s failed with the status code %d and the following text: %s",
-                    url,
-                    response.status_code,
-                    response.text,
-                )
-                raise CachitoError(
-                    "Could not reach the Cachito API to find the requests to be marked as stale"
-                )
-
-            json_response = response.json()
-            identify_and_mark_stale_requests(json_response["items"])
-            url = json_response["meta"]["next"]
+        maybe_stale_requests = find_all_requests_in_state(state)
+        identify_and_mark_stale_requests(maybe_stale_requests)
+
+
+def find_all_requests_in_state(state):
+    """
+    Find all requests in specified state.
+
+    :param str state: state of request, e.g. 'complete', 'in_progress'
+    :return: list of Cachito requests (as JSON data) in specified state
+    """
+    found_requests = []
+
+    url = f"{config.cachito_api_url.rstrip('/')}/requests"
+    while url:
+        try:
+            response = session.get(url, params={"state": state}, timeout=config.cachito_api_timeout)
+        except requests.RequestException:
+            msg = f"The connection failed when querying {url}"
+            log.exception(msg)
+            raise CachitoError(msg)
+
+        if not response.ok:
+            log.error(
+                "The request to %s failed with the status code %d and the following text: %s",
+                url,
+                response.status_code,
+                response.text,
+            )
+            raise CachitoError(
+                "Could not reach the Cachito API to find the requests to be marked as stale"
+            )
+
+        json_response = response.json()
+        found_requests.extend(json_response["items"])
+        url = json_response["meta"]["next"]
+
+    return found_requests
```

Comment:
Should we ensure entries in `found_requests` are unique (using ther `id`s)? I suppose the contents in each page may be quite dynamic depending on the load in Chachito at a given hour.

This is a minor issue since the request will not be update if we POST the same data, but still, less requests...

However, it does make the API throw a (somewhat unrelated) error when applying the operation twice in the same request:

```
[2020-10-13 09:42:41 flask.app INFO auth.load_user_from_request] The REMOTE_USER environment variable wasn't set on the request, but the LOGIN_DISABLED configuration is set to True.
[2020-10-13 09:42:41 flask.app ERROR api_v1.patch_request] Failed to delete the log file /var/log/cachito/requests/10.log
Traceback (most recent call last):
  File "/src/cachito/web/api_v1.py", line 431, in patch_request
    os.remove(path_to_file)
FileNotFoundError: [Errno 2] No such file or directory: '/var/log/cachito/requests/10.log'
```

Reply:
Good point, fixed